src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.serviceimpl.FruitTransactionValidatorImpl;
+import java.util.Optional;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private FruitTransactionValidatorImpl validator;
```

Comment:
The `validator` field is not initialized. It's a good practice to initialize such dependencies through a constructor to make the code more testable and to follow the Dependency Inversion principle.

Reply:
valid point

Reply:
not fixed